src/indexes.rs

```diff
@@ -464,7 +422,7 @@ impl Index {
     /// # movie_index.delete().await.unwrap().wait_for_completion(&client, None, None).await.unwrap();
     /// # });
     /// ```
-    pub async fn add_or_update<T: Document>(
+    pub async fn add_or_update<'a ,T: DeserializeOwned + Serialize + std::fmt::Debug>(
```

Comment:
```suggestion
    pub async fn add_or_update<T: DeserializeOwned + Serialize + std::fmt::Debug>(
```

Same, `'a` is not used I think.